src/main/java/com/amazonaws/services/kinesis/multilang/MultiLangDaemon.java

```diff
@@ -140,11 +140,27 @@ public static void main(String[] args) {
         ExecutorService executorService = config.getExecutorService();
 
         // Daemon
-        MultiLangDaemon daemon = new MultiLangDaemon(
+        final MultiLangDaemon daemon = new MultiLangDaemon(
                 config.getKinesisClientLibConfiguration(),
                 config.getRecordProcessorFactory(),
                 executorService);
 
+        Runtime.getRuntime().addShutdownHook(new Thread()
+        {
+            @Override
+            public void run()
+            {
+                LOG.info("Process terminanted, will initiate shutdown.");
+                try {
+                    Future<Void> fut = daemon.worker.requestShutdown();
+                    fut.get();
```

Comment:
In a worst case scenario this could cause the shutdown to become stuck.  It would be best to start with some timeout value of how long to wait for the shutdown request to complete.  I usually start with something in the range of 5 seconds.  In the future it may make sense to make it configurable.